### DISCUSSION TITLE
The boundary conditions conflict with the mirror symmetry

### DISCUSSION THREAD

**Discussion by FGmagi** [NanoComp/meep#2893] [Q&A]
I designed a model involving mirror symmetry along the XY axes:
```
cell.sim.symmetries.append(mp.Mirror(direction=mp.X, phase=1))
cell.sim.symmetries.append(mp.Mirror(direction=mp.Y, phase=1))
```

With Bloch-periodic boundaries in the XY directions:
`k_point = mp.Vector3(0, 0, 0)`

And metallic boundaries in the Z direction:
```

cell.sim.set_boundary(side=mp.High, direction=mp.Z, condition=mp.Metallic)
cell.sim.set_boundary(side=mp.Low, direction=mp.Z, condition=mp.Metallic)
```

However, I found that after adding the metallic boundaries, the mirror symmetry conditions no longer work. Specifically, the runtime increased by about four times, and the observed output electric field indicates that only one-quarter of the structure is being simulated.

I would appreciate any insights or solutions.

**Comment by stevengj** [comment]
Is the field diverging?  Sometimes we've observed an instability arising when multiple mirror symmetry planes are combined in 3d, IIRC.

See https://github.com/NanoComp/meep/issues/132

**Comment by mdfeinstein** [comment]
I'm not sure this addresses your stated problem, but I noticed that your symmetries seem potentially problematic. You have both phases as 1, but shouldn't one of the phases be -1? see this [example for instance](https://meep.readthedocs.io/en/latest/Python_Tutorials/Basics/#mie-scattering-of-a-lossless-dielectric-sphere), where the electric field component is along the Z axis (propagating along X) and the symmetries are set as:
`symmetries = [mp.Mirror(mp.Y),
              mp.Mirror(mp.Z,phase=-1)]`

**Comment by oskooi** [comment]
This issue may be fixed by #3155.
